From e3b3e53f72647fa189ae245d73dc560166bb57d3 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Sat, 24 Jan 2015 21:12:02 +0100
Subject: [PATCH 190/506] commands/macbless: Fix potential overflow.

Is a minor concern as no such FS would be created under normal circumstances
and failure was benign.

Found by: Coverity scan.
---
 grub-core/commands/macbless.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/commands/macbless.c b/grub-core/commands/macbless.c
index cf90eea..f9e01b6 100644
--- a/grub-core/commands/macbless.c
+++ b/grub-core/commands/macbless.c
@@ -106,7 +106,7 @@ grub_mac_bless_inode (grub_device_t dev, grub_uint32_t inode, int is_dir,
       ablk_size = grub_be_to_cpu32 (volheader.hfs.blksz);
       ablk_start = grub_be_to_cpu16 (volheader.hfs.first_block);
       embedded_offset = (ablk_start
-			 + extent_start
+			 + ((grub_uint64_t) extent_start)
 			 * (ablk_size >> GRUB_DISK_SECTOR_BITS));
 
       err =
-- 
2.4.3

